Templates/AppSource App/.github/workflows/UpdateGitHubGoSystemFiles.yaml

```diff
@@ -74,18 +74,44 @@ jobs:
 
       - name: Calculate Input
         env:
-          directCommit: '${{ github.event.inputs.directCommit }}'
+          directCommit: ${{ github.event.inputs.directCommit }}
           downloadLatest: ${{ github.event.inputs.downloadLatest }}
           eventName: ${{ github.event_name }}
+          GhTokenWorkflow: ${{ fromJson(steps.ReadSecrets.outputs.Secrets).ghTokenWorkflow }}
         run: |
-          $errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+          $errorActionPreference = "Stop";
+          $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
           $directCommit = $ENV:directCommit
           $downloadLatest = $ENV:downloadLatest
           Write-Host $ENV:eventName
           if ($ENV:eventName -eq 'schedule') {
-            Write-Host "Running Update AL-Go System Files on a schedule. Setting DirectCommit and DownloadLatest to true"
-            $directCommit = 'true'
             $downloadLatest = 'true'
+            $directCommit = 'true'
+            try {
+              Write-Host "Running Update AL-Go System Files on a schedule."
+              $ENV:GH_TOKEN = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($env:GhTokenWorkflow))
+              $rules = gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/$env:GITHUB_REPOSITORY/rules/branches/$env:GITHUB_REF_NAME | ConvertFrom-Json
+              $branchProtection = gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/$env:GITHUB_REPOSITORY/branches/$env:GITHUB_REF_NAME/protection | ConvertFrom-Json
+              if ($rules | Where-Object { $_.type -eq 'pull_request' }) {
```

Comment:
Generally, rulesets also allow bypassing the rules:
![image](https://github.com/microsoft/AL-Go/assets/43066499/7943cfc0-4e20-4715-9010-d28520a38fa4)

But I don't think there is a way to check if the `ghTokenWorkflow` is from an account in the bypass list.

Reply:
Implemented Alexanders idea